src/PSR7/Validators/PathValidator.php

```diff
@@ -39,14 +40,21 @@ public function validate(OperationAddress $addr, MessageInterface $message) : vo
     /**
      * @throws InvalidPath
      * @throws NoPath
+     * @throws NoParameter
      */
     private function validateRequest(OperationAddress $addr, RequestInterface $message) : void
     {
         $specs = $this->finder->findPathSpecs($addr);
 
         $path             = $message->getUri()->getPath();
         $pathParsedParams = $addr->parseParams($path); // ['id'=>12]
-        $validator        = new SchemaValidator($this->detectValidationStrategy($message));
+        if ($pathParsedParams===[]) {
+            return;
+        }
+        if ($specs === []) {
+            throw NoParameter::fromPath($addr->path());
```

Comment:
This tends to be not a `ValidationFailed` exception, but rather and `InvalidSchema` since every parameter in path should be described. But speaking more general passing this point means that we are validating address which doesn't match any operation in our spec and there is nothing to do with missing params.